### PR TITLE
Unread count

### PR DESCRIPTION
**Pull request by elverkilde**
Simply adds the unread count and last message at parsing.

### CODE REVIEW

File: tests/integration/main.js [hunk 1]
```diff
@@ -821,6 +821,21 @@ test(`send messages [sends two messages to Bob's room]`, t => {
   t.timeoutAfter(TEST_TIMEOUT)
 })
 
+test(`check unread messages [Bob has 2 in Bob's room]`, t => {
+  const expectedRoomCounts = [(alicesRoom.id, 0), (bobsRoom.id, 2)].sort()
```
Comment (callum-oakley) [inline tests/integration/main.js]: tuples: not a thing!

```js
(x, y) === y
```

[god only knows what they were thinking with this one](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Comma_Operator)

just changing `(...)` to `[...]` should do the trick though

File: tests/integration/main.js [hunk 2]
```diff
@@ -821,6 +821,21 @@ test(`send messages [sends two messages to Bob's room]`, t => {
...
+  fetchUser(t, "bob")
+    .then(bob => {
+      t.deepEqual(
+        map(r => (r.id, r.unreadCount), bob.rooms).sort(),
```
Comment (callum-oakley) [inline tests/integration/main.js]: and here

File: tests/integration/main.js (hunk 1 above)
Comment (elverkilde) [reply tests/integration/main.js]: :+1:

Comment (elverkilde) [reply tests/integration/main.js]: changed it to filter on bobs room as we don't have a reference to the other room he's in.

**Comment by callum-oakley** [comment]
Approved for merging in to the `v4` branch. Don't merge to master just yet please!

**Pull request merged**